hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/DynamicConfigVersionTest.java

```diff
@@ -22,51 +22,46 @@
 import com.hazelcast.config.ListenerConfig;
 import com.hazelcast.config.SplitBrainProtectionConfig;
 import com.hazelcast.config.WanReplicationConfig;
-import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
-import org.junit.Test;
 import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 
 import java.lang.reflect.Method;
-import java.util.HashSet;
+import java.util.Arrays;
 import java.util.Set;
+import java.util.stream.Stream;
 
-import static com.hazelcast.internal.dynamicconfig.ClusterWideConfigurationService.CONFIG_TO_VERSION;
-import static java.lang.String.format;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
-@RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class DynamicConfigVersionTest {
+    /** config classes not supported by dynamic data structure config */
+    private static Set<Class<?>> nonDynamicConfigClasses;
 
-    // config classes not supported by dynamic data structure config
-    private static final Set<Class<?>> NON_DYNAMIC_CONFIG_CLASSES;
+    @BeforeAll
+    public static void setUp() {
+        nonDynamicConfigClasses = Set.of(WanReplicationConfig.class, SplitBrainProtectionConfig.class, ListenerConfig.class,
+                DeviceConfig.class, DynamicConfigurationConfig.class);
+    }
```

Comment:
I feel this was better left as a statically initialized value (could still leverage `Set.of()`)

Reply:
> I feel this was better left as a statically initialized value (could still leverage `Set.of()`)

Can I ask why? My understanding of JUnit is that you want to ensure that everything is initialized via a `@Before`, both for testabilities sake, but also for performance - something about allowing JUnit to decide when they are initialized.

Reply:
That makes sense for non-static variables, but in this case the `Set` contents are never going to change and they're always needed for the test, so initialization order is not going to change anything really. In terms of performance, having it statically configured will give the compiler more room to optimize, whereas embedding within a method does not (on top of the minute performance overhead of the method calls).

Reply:
I also just noticed `WanReplicationConfig` is actually dynamic now, so this list needs updating (I believe @gbarnett-hz added dynamic support previously)